fpr/pipelines/nodejs_metadata.py

```diff
@@ -274,31 +297,89 @@ def group_by_org_repo_ref_path(
             yield (org_repo_key, ref_value_key, dep_file_parent_key), file_rows
 
 
-async def run_pipeline(
-    source: Generator[Dict[str, Any], None, None], args: argparse.Namespace
-) -> AsyncGenerator[Dict, None]:
-    log.info(f"{pipeline.name} pipeline started")
+def iter_task_envs(
+    args: argparse.Namespace
+) -> Generator[
+    Tuple[Language, PackageManager, DockerImage, ChainMap, List[ContainerTask]],
+    None,
+    None,
+]:
+    enabled_languages = args.language or language_names
+    if not args.language:
+        log.debug(f"languages not specified using all of {enabled_languages}")
+
+    enabled_package_managers = args.package_manager or package_manager_names
+    if not args.package_manager:
+        log.debug(
+            f"package managers not specified using all of {enabled_package_managers}"
+        )
 
-    # TODO: build images for each lang version, etc? (configure with CLI args?)
-    try:
-        await build_container()
-    except Exception as e:
-        log.error(
-            f"error occurred building the nodejs metadata image: {e}\n{exc_to_str()}"
+    enabled_image_names = args.docker_image or docker_image_names
+    if not args.docker_image:
+        log.debug(
+            f"docker image names not specified using all of {enabled_image_names}"
+        )
+
+    for language_name, package_manager_name, image_name in itertools.product(
+        enabled_languages, enabled_package_managers, enabled_image_names
+    ):
+        if language_name not in languages:
+            continue
+        language = languages[language_name]
+        if package_manager_name not in language.package_managers:
+            continue
+        package_manager = language.package_managers[package_manager_name]
+        if image_name not in language.images:
+            continue
+        image = language.images[image_name]
+
+        version_commands = ChainMap(
+            language.version_commands,
+            *[pm.version_commands for pm in language.package_managers.values()],
```

Comment:
NB: This assumes all package managers are installed in the lang image (might not be the case for python).